module/zfs/zil.c

```diff
@@ -1878,6 +1880,8 @@ zil_lwb_write_issue(zilog_t *zilog, lwb_t *lwb)
 	 */
 	memset(lwb->lwb_buf + lwb->lwb_nused, 0, wsz - lwb->lwb_nused);
 
+	spa_config_enter(zilog->zl_spa, SCL_STATE, lwb, RW_READER);
```

Comment:
if we can, it might be good to write a comment w.r.t. why we need to grab the lock here...

Reply:
That's a good idea. I will submit a follow-on PR so that others don't step on this landmine.